Druid/DRCogitCanonicaliser.class.st

```diff
@@ -24,6 +24,42 @@ DRCogitCanonicaliser >> applyTo: cfg [
 		worklist removeLast acceptVisitor: self ]
 ]
 
+{ #category : 'visiting' }
+DRCogitCanonicaliser >> collapseBitwiseOperationIfPossible: aDRGetConditionCode [
+	"Convert TestEqualsThan(BitAnd(a, b), 0)     into  TestNoMask(a, b)
+	     and TestNotEqualsThan(BitAnd(a, b), 0)  into  TestMask(a, b)"
+
+	| oldCondition newCondition newInstruction |
+	oldCondition := aDRGetConditionCode condition.
+	(oldCondition isEqualsThanComparison or:
+		 oldCondition isNotEqualsThanComparison) ifFalse: [ ^ aDRGetConditionCode ].
+
+	"Move constant, if present, to the right"
+	aDRGetConditionCode operand1 isConstant ifTrue: [
+		aDRGetConditionCode operands:
+			{ oldCondition }, aDRGetConditionCode operands reverse ].
```

Comment:
We should `reverse` the condition too. I understand that here is the same, because we are checking by _equality comparison_ and it is commutative.
I think that it's better to be explicit:
```suggestion
	"Move constant, if present, to the right"
	aDRGetConditionCode operand1 isConstant ifTrue: [
		aDRGetConditionCode operands:
			{ oldCondition reversed }, aDRGetConditionCode operands reverse ].
```

Reply:
Yeah, makes sense